### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-7

### ISSUE THREAD

**Issue by vlotorev** [useblocks/sphinxcontrib-needs#86]
Hi,
I'm using 0.3.10 version, and python 2.7.
Consider having following code with non ASCII characters in need title:
```
.. req:: Загрузка CPU
   :id: RQ001

Linking to requirement: :need:`RQ001`
```
Compiling it fails with exception:
```
  File "/home/.../lib/python2.7/site-packages/sphinxcontrib/needs/roles/need_ref.py", line 46, in process_need_ref
    title = title if len(title) < 30 else '{}...'.format(title[:27])
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-7: ordinal not in range(128)
```
It can be fixed by changing [line](https://github.com/useblocks/sphinxcontrib-needs/blob/master/sphinxcontrib/needs/roles/need_ref.py#L45):
` title = title if len(title) < 30 else '{}...'.format(title[:27])`
to
` title = title if len(title) < 30 else u'{}...'.format(title[:27])`

After that new exception arises:
```
  File "/home/.../lib/python2.7/site-packages/sphinxcontrib/needs/roles/need_ref.py", line 58, in process_need_ref
    links_back=";".join(target_need["links_back"]))
UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-7: ordinal not in range(128)
```
Which can be fixed changing [line](https://github.com/useblocks/sphinxcontrib-needs/blob/master/sphinxcontrib/needs/roles/need_ref.py#L47):
`link_text = app.config.needs_role_need_template.format(title=title,`
to
`link_text = unicode(app.config.needs_role_need_template).format(title=title,`

Totally sure the last fix is not fully proper as `app.config.needs_role_need_template` should unicodized at the place it was created.

**Comment by danwos** [comment]
Hi,
thanks for the detailed report.
I have uploaded a fix. Do you have some time to support with a rough test of this fix?

I everything is fine I will create a new release for pypi.

**Comment by vlotorev** [comment]
Hi,  thanks for the fix.
I've checked commit https://github.com/useblocks/sphinxcontrib-needs/commit/549b948ecbd344064576ca6162896fbf5624f1e8 and now HTML build doesn't fail.

**Comment by danwos** [comment]
Released in 0.3.11. Sorry for the delay, I wanted to add some more features to the release...
